homeassistant/components/intellifire/__init__.py

```diff
@@ -32,79 +43,127 @@
-        await cloud_api.login(
-            username=entry.data[CONF_USERNAME],
-            password=entry.data[CONF_PASSWORD],
+
+        # See if we can find the fireplace first by serial and then secondly by IP.
+
+        serial = config_entry.title.replace("Fireplace ", "")
+
+        # If serial matches the hex style pattern we'll assume its good
+        valid_serial = bool(re.match(r"^[0-9A-Fa-f]{32}$", serial))
+
+        new_data = (
+            cloud_interface.user_data.get_data_for_serial(serial)
+            if valid_serial
+            else None
         )
-        api_key = cloud_api.get_fireplace_api_key()
-        user_id = cloud_api.get_user_id()
-        # Update data entry
+        if not new_data:
+            new_data = cloud_interface.user_data.get_data_for_ip(new[CONF_IP_ADDRESS])
+
+    if not new_data:
+        raise ConfigEntryAuthFailed
+
+    # Find the correct fireplace
+    if new_data is not None:
+        new[CONF_API_KEY] = new_data.api_key
+        new[CONF_WEB_CLIENT_ID] = new_data.web_client_id
+        new[CONF_AUTH_COOKIE] = new_data.auth_cookie
+
+        new[CONF_IP_ADDRESS] = new_data.ip_address
+        new[CONF_SERIAL] = new_data.serial
+
+        config_entry.version = 1
         hass.config_entries.async_update_entry(
-            entry,
-            data={
-                **entry.data,
-                CONF_API_KEY: api_key,
-                CONF_USER_ID: user_id,
-            },
+            config_entry,
+            data=new,
+            options={CONF_READ_MODE: "local", CONF_CONTROL_MODE: "local"},
+            unique_id=serial,
         )
+        LOGGER.debug("Pseudo Migration %s successful", config_entry.version)
 
-    else:
-        api_key = entry.data[CONF_API_KEY]
-        user_id = entry.data[CONF_USER_ID]
+    return config_entry
 
-    # Instantiate local control
-    api = IntellifireAPILocal(
-        fireplace_ip=entry.data[CONF_HOST],
-        api_key=api_key,
-        user_id=user_id,
-    )
 
-    # Define the update coordinator
-    coordinator = IntellifireDataUpdateCoordinator(
-        hass=hass,
-        api=api,
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up IntelliFire from a config entry."""
+    LOGGER.debug("Setting up config entry: %s", entry.unique_id)
+
+    if CONF_USERNAME not in entry.data:
+        LOGGER.debug("Super Old config entry format detected: %s", entry.unique_id)
+        raise ConfigEntryAuthFailed
+
+    if CONF_IP_ADDRESS not in entry.data:
+        LOGGER.debug("Old config entry format detected: %s", entry.unique_id)
+        entry = await _async_pseudo_migrate_entry(hass, entry)
+
+    # Fireplace will throw an error if it can't connect
+    try:
+        fireplace: UnifiedFireplace = (
+            await UnifiedFireplace.build_fireplace_from_common(
+                _construct_common_data(entry)
+            )
+        )
+        LOGGER.info("Waiting for Fireplace to Initialized")
```

Comment:
```suggestion
        LOGGER.info("Waiting for Fireplace to Initialize")
```

Also, should there be a success-print too?